inference-engine/src/transformations/src/transformations/common_optimizations/interpolate_sequence_fusion.cpp

```diff
@@ -0,0 +1,196 @@
+    std::set<int64_t> fst_axes_set(fst_axes_vector.begin(), fst_axes_vector.end());
+    for (const auto& a : snd_axes_vector) {
+        if (fst_axes_set.count(a) != 0) return false;
+    }
+    return true;
+}
+
+bool shape_calculation_mode_can_use_constant_inputs(const std::shared_ptr<opset8::Interpolate>& interpolate) {
+    const auto& attrs = interpolate->get_attrs();
+    if (attrs.shape_calculation_mode == ngraph::opset8::Interpolate::ShapeCalcMode::SIZES) {
+        return std::dynamic_pointer_cast<opset8::Constant>(interpolate->input_value(1).get_node_shared_ptr()) != nullptr;
+    }
+    return std::dynamic_pointer_cast<opset8::Constant>(interpolate->input_value(2).get_node_shared_ptr()) != nullptr;
+}
+
+bool is_candidate_for_fusion(const std::shared_ptr<opset8::Interpolate>& interpolate) {
+    return (interpolate->get_input_partial_shape(0).rank().is_static()) &&
+           (interpolate->inputs().size() != 4 || std::dynamic_pointer_cast<opset8::Constant>(interpolate->input_value(3).get_node_shared_ptr())) &&
+           shape_calculation_mode_can_use_constant_inputs(interpolate);
+}
+
+std::vector<int64_t> get_interpolated_axes(const std::shared_ptr<opset8::Interpolate>& interpolate) {
+    if (interpolate->inputs().size() != 4) {
+        const auto input_rank = interpolate->get_input_partial_shape(0).rank().get_length();
+
+        std::vector<int64_t> default_value(input_rank);
+        std::iota(default_value.begin(), default_value.end(), 0);
+
+        return default_value;
+    }
+    return std::dynamic_pointer_cast<opset8::Constant>(interpolate->input_value(3).get_node_shared_ptr())->cast_vector<int64_t>();
+}
+
+bool can_be_fused(const std::shared_ptr<opset8::Interpolate>& fst, const std::shared_ptr<opset8::Interpolate>& snd) {
+    // The first Interpolate (fst) must have only one consumer.
+    for (const auto& output : fst->outputs()) {
+        for (const auto& consumer : output.get_target_inputs()) {
+            if (consumer.get_node() != snd.get()) return false;
+        }
+    }
+
+    if (!compatible_attrs(fst->get_attrs(), snd->get_attrs()) || !is_candidate_for_fusion(fst) || !is_candidate_for_fusion(snd)) return false;
+
+    const auto fst_axes = get_interpolated_axes(fst);
+    const auto snd_axes = get_interpolated_axes(snd);
+    return compatible_axes(fst_axes, snd_axes);
+}
+
+ngraph::NodeVector subgraph_for_sizes_calculation_mode(const std::shared_ptr<opset8::Interpolate>& fst, const std::shared_ptr<opset8::Interpolate>& snd) {
+    const auto fst_axes = get_interpolated_axes(fst);
+    const auto snd_axes = get_interpolated_axes(snd);
+    const auto fst_sizes = std::dynamic_pointer_cast<opset8::Constant>(fst->input_value(1).get_node_shared_ptr())->cast_vector<int64_t>();
+    const auto snd_sizes = std::dynamic_pointer_cast<opset8::Constant>(snd->input_value(1).get_node_shared_ptr())->cast_vector<int64_t>();
```

Comment:
You always need to check the result of dynamic_pointer_cast for nullpt.

Reply:
Fixed.